contrastive_vi/model/total_contrastive_vi.py

```diff
@@ -0,0 +1,748 @@
+            msg = (
+                "Some proteins have all 0 counts in some batches. "
+                + "These proteins will be treated as missing measurements; however, "
+                + "this can occur due to experimental design/biology. "
+                + "Reinitialize the model with `override_missing_proteins=True`,"
+                + "to override this behavior."
+            )
+            warnings.warn(msg, UserWarning)
+            self._use_adversarial_classifier = True
+        else:
+            batch_mask = None
+            self._use_adversarial_classifier = False
+
+        emp_prior = (
+            empirical_protein_background_prior
+            if empirical_protein_background_prior is not None
+            else (self.summary_stats.n_proteins > 10)
+        )
+        if emp_prior:
+            prior_mean, prior_scale = self._get_totalvi_protein_priors(adata)
+        else:
+            prior_mean, prior_scale = None, None
+
+        n_cats_per_cov = (
+            self.adata_manager.get_state_registry(REGISTRY_KEYS.CAT_COVS_KEY)[
+                CategoricalJointObsField.N_CATS_PER_KEY
+            ]
+            if REGISTRY_KEYS.CAT_COVS_KEY in self.adata_manager.data_registry
+            else None
+        )
+
+        n_batch = self.summary_stats.n_batch
+        use_size_factor_key = (
+            REGISTRY_KEYS.SIZE_FACTOR_KEY in self.adata_manager.data_registry
+        )
+        library_log_means, library_log_vars = None, None
+        if not use_size_factor_key:
+            library_log_means, library_log_vars = _init_library_size(
+                self.adata_manager, n_batch
+            )
+
+        self.module = TotalContrastiveVIModule(
+            n_input_genes=self.summary_stats["n_vars"],
+            n_input_proteins=self.summary_stats["n_proteins"],
+            n_batch=n_batch,
+            n_background_latent=n_background_latent,
+            n_salient_latent=n_salient_latent,
+            n_continuous_cov=self.summary_stats.get("n_extra_continuous_covs", 0),
+            n_cats_per_cov=n_cats_per_cov,
+            gene_dispersion=gene_dispersion,
+            protein_dispersion=protein_dispersion,
+            gene_likelihood=gene_likelihood,
+            latent_distribution=latent_distribution,
+            protein_batch_mask=batch_mask,
+            protein_background_prior_mean=prior_mean,
+            protein_background_prior_scale=prior_scale,
+            use_size_factor_key=use_size_factor_key,
+            library_log_means=library_log_means,
+            library_log_vars=library_log_vars,
+            **model_kwargs,
+        )
+        self._model_summary_string = "totalContrastiveVI."
+        # Necessary line to get params to be used for saving and loading.
+        self.init_params_ = self._get_init_params(locals())
+        logger.info("The model has been initialized")
+
+    @classmethod
+    @setup_anndata_dsp.dedent
+    def setup_anndata(
```

Comment:
The implementation of this function is quite different from the original due to the need to use the `AnnDataManager` class in the newer versions of scvi-tools.